service/pkg/service/worker.go

```diff
@@ -76,18 +79,23 @@ func (w *Worker) Start() error {
 		}
 		logger.Debugw("Request claimed", "ID", req.Id)
 
-		err = w.Run(req, key, start, stop)
+		err = w.Run(req, start, stop)
 		if err != nil {
 			logger.Errorw("Recorder failed", err)
 			return err
 		}
+
+		if w.shutdown {
```

Comment:
if this is being accessed from different goroutines, you'd want an atomic or lock around it.